src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,122 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.excepttion.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturerQuery = "INSERT INTO manufacturers(name, country) values(?, ?);";
+        try (
+                Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertManufacturerStatement =
+                        connection.prepareStatement(insertManufacturerQuery,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            insertManufacturerStatement.setString(1, manufacturer.getName());
+            insertManufacturerStatement.setString(2, manufacturer.getCountry());
+            insertManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = insertManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                manufacturer.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert manufacturer to database", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateManufacturerQuery =
+                "UPDATE manufacturers SET name = (?), country = (?)"
+                        + "WHERE id = (?) AND is_deleted = FALSE;";
+        try (
+                Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(updateManufacturerQuery)
+                ) {
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.setLong(3, manufacturer.getId());
+            preparedStatement.executeUpdate();
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update "
+                    + manufacturer.toString() + "in database", e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        try (
+                Connection connection = ConnectionUtil.getConnection();
+                Statement getAllManufacturer = connection.createStatement()) {
+            ResultSet resultSet = getAllManufacturer
+                    .executeQuery("SELECT * FROM manufacturers WHERE is_deleted = FALSE");
+            while (resultSet.next()) {
+                Manufacturer manufacturer = getManufacturerFromResultSet(resultSet);
+                manufacturers.add(manufacturer);
```

Comment:
we can inline this operation
```suggestion
                manufacturers.add(getManufacturerFromResultSet(resultSet));
```